src/main/kotlin/org/jitsi/jicofo/CodecConfig.kt

```diff
@@ -0,0 +1,109 @@
+/*
+ * Jicofo, the Jitsi Conference Focus.
+ *
+ * Copyright @ 2020 - present 8x8, Inc
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.jitsi.jicofo
+
+import org.jitsi.config.JitsiConfig
+import org.jitsi.metaconfig.config
+
+class VideoCodecConfig(
+    legacyBase: String,
+    newBase: String,
+    private val codecName: String
+) {
+    private val enabled: Boolean by config {
+        "$legacyBase.ENABLE_$codecName".from(JitsiConfig.legacyConfig)
+        "$newBase.enabled".from(JitsiConfig.newConfig)
+    }
+
+    fun enabled() = enabled && pt > 0
+
+    private val pt: Int by config {
+        "$legacyBase.${codecName}_PT".from(JitsiConfig.legacyConfig)
+        "$newBase.pt".from(JitsiConfig.newConfig)
+    }
+
+    @Throws(IllegalStateException::class)
+    fun pt(): Int = if (enabled()) pt else throw IllegalStateException("$codecName is not enabled.")
+
+    private val rtxPt: Int by config {
+        "$legacyBase.${codecName}_RTX_PT".from(JitsiConfig.legacyConfig)
+        "$newBase.rtx-pt".from(JitsiConfig.newConfig)
+    }
+
+    fun rtxEnabled(): Boolean = enabled() && rtxPt > 0
+
+    @Throws(IllegalStateException::class)
+    fun rtxPt(): Int = if (rtxEnabled()) rtxPt else throw IllegalStateException("RTX is not enabled for $codecName")
+
+}
+
+open class RtpExtensionConfig(
+    legacyEnabledName: String,
+    protected val newBase: String
+) {
+    open val enabled: Boolean by config {
+        legacyEnabledName.from(JitsiConfig.legacyConfig)
+        "$newBase.enabled".from(JitsiConfig.newConfig)
+    }
+    fun enabled() = enabled
+
+    val id: Int by config {
+        "$newBase.id".from(JitsiConfig.newConfig)
+    }
+    fun id() = id
+}
+
+class TccConfig : RtpExtensionConfig("none", "${CodecConfig.EXTENSIONS_BASE}.tcc") {
+    override val enabled: Boolean by config {
+        "$newBase.enabled".from(JitsiConfig.newConfig)
```

Comment:
I'm confused by this...only TCC doesn't check for a legacy name?

Reply:
There was no legacy config for TCC.